vars/withKubicEnvironment.groovy

```
@@ -66,9 +67,10 @@ def call(Map parameters = [:], Closure body) {
             // Destroy the Kubic Environment
             stage('Destroy Environment') {
                 try {
-                    echo "TODO"
+                    cleanupEnvironment()
                 } catch (Exception exc) {
                     echo "Failed to Destroy Environment"
+                    throw exc
```

If we rethrow, the logs will not be archived - so we have to be careful here to log it, but not let it bubble up